waku/v2/protocol/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -196,7 +196,16 @@ proc backfillRootQueue*(g: OnchainGroupManager, blockTable: BlockTable): Future[
         # add the backfilled root
         g.validRoots.addLast(g.validRootBuffer.popLast())
 
-proc getEvents*(g: OnchainGroupManager, fromBlock: BlockNumber, toBlock: Option[BlockNumber] = none(BlockNumber)): Future[BlockTable] {.async.} =
+proc insert(blockTable: var BlockTable, blockNumber: BlockNumber, member: Membership) =
+  if blockTable.hasKeyOrPut(blockNumber, @[member]):
+    try:
+      blockTable[blockNumber].add(member)
+    except KeyError: # qed
+      error "could not insert member into block table", blockNumber=blockNumber, member=member
```

Comment:
Would an `error` log here be enough error handling (vs. e.g. adding metrics returning an `err` `Result`, etc.)?

Reply:
the `hasKeyOrPut` takes care of the error. nim throws an error without having context of the previous line (thats why an error log is OK imo in this case)